diff --git a/node_modules/@docusaurus/core/lib/commands/start.js b/node_modules/@docusaurus/core/lib/commands/start.js
index 88040f2..536d89d 100644
--- a/node_modules/@docusaurus/core/lib/commands/start.js
+++ b/node_modules/@docusaurus/core/lib/commands/start.js
@@ -145,6 +145,7 @@ async function start(siteDir, cliOptions) {
             stats: 'summary',
         },
         static: {
+            publicPath: baseUrl,
             directory: path_1.default.resolve(siteDir, constants_1.STATIC_DIR_NAME),
             watch: {
                 // Useful options for our own monorepo using symlinks!
